gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/NonLocalRegionEntry.java

```diff
@@ -46,7 +47,7 @@
 import com.gemstone.gemfire.internal.offheap.annotations.Released;
 import com.gemstone.gemfire.internal.shared.Version;
 
-public class NonLocalRegionEntry implements RegionEntry, VersionStamp {
+public class NonLocalRegionEntry implements RegionEntry, VersionStamp, DataSerializable {
```

Comment:
Why implementing DataSerializable required? It already implements DataSerializableFixedID no? But I saw the code and in that it is not registered as a fixed id class in DSFIDFactory but then why does it implement toData and fromData? Could not understand.
And I guess DataSerializable is more costly than fixedId thing. It writes the full classname during serialization? Please check if fixed id will be better.